tests/python_tests/test_generate_api.py

```diff
@@ -0,0 +1,119 @@
+# Copyright (C) 2023-2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import pytest
+from list_test_models import models_list
+
+
+@pytest.fixture(scope="module", params=models_list())
+def model_fixture(request):
+    model_id, path = request.param
+    from transformers import AutoTokenizer, AutoModelForCausalLM
+    tokenizer = AutoTokenizer.from_pretrained(model_id)
+    model = AutoModelForCausalLM.from_pretrained(model_id)
+    return model_id, path, tokenizer, model
+
+def run_hf_ov_genai_comparison(model_fixture, generation_config, prompt):
+    import openvino_genai as ov_genai
+    model_id, path, tokenizer, model = model_fixture
+
+    generation_config_hf = generation_config.copy()
+    # in OpenVINO GenAI this parameter is called stop_criteria,
+    # while in HF it's called early_stopping. 
+    # HF values True, False and "never" correspond to OV GenAI values "early", "heuristic" and "never"
+    if generation_config_hf.get('stop_criteria'):
+        generation_config_hf['early_stopping'] = stop_criteria_map()[generation_config_hf.pop('stop_criteria')]
+
+    encoded_prompt = tokenizer.encode(prompt, return_tensors='pt', add_special_tokens=True)
+    hf_encoded_output = model.generate(encoded_prompt, **generation_config_hf)
+    hf_output = tokenizer.decode(hf_encoded_output[0, encoded_prompt.shape[1]:])
+
+    device = 'CPU'
+    # pipe = ov_genai.LLMPipeline(path, device)
+    
+    import os
+    build_dir = os.getenv('GENAI_BUILD_DIR', 'build')
+    ov_tokenizers_path = f'{build_dir}/openvino_tokenizers/src/'
```

Comment:
can we use relative path instead?
E.g. tokenizers, genai and python files are copies to the same directory where we can find tokenizer.so relatively to python files.

Reply:
Yes. To do the same in C++ I'd need a path to the current library. Does openvino have have a function for that or an implementation I could copy?

Reply:
https://github.com/openvinotoolkit/openvino/blob/master/src/common/util/src/file_util.cpp#L489